app/ide-desktop/lib/dashboard/src/modals/TermsOfServiceModal.tsx

```diff
@@ -0,0 +1,145 @@
+/**
+ * @file
+ *
+ * Modal for accepting the terms of service.
+ */
+
+import * as React from 'react'
+
+import * as reactQuery from '@tanstack/react-query'
+import * as router from 'react-router'
+import invariant from 'tiny-invariant'
+
+import * as authProvider from '#/providers/AuthProvider'
+import * as localStorageProvider from '#/providers/LocalStorageProvider'
+import * as textProvider from '#/providers/TextProvider'
+
+import * as aria from '#/components/aria'
+import * as ariaComponents from '#/components/AriaComponents'
+import Checkbox from '#/components/styled/Checkbox'
+
+import type * as localStorage from '#/utilities/LocalStorage'
+import LocalStorage from '#/utilities/LocalStorage'
+
+declare module '#/utilities/LocalStorage' {
+  /**
+   * The data stored in the `termsOfService` key.
+   */
+  interface AcceptedTosData {
+    /**
+     * The hash of the latest terms of service version that the user has accepted.
+     */
+    readonly versionHash: string
+  }
+
+  /**
+   * Contains the latest terms of service version hash that the user has accepted.
+   */
+  interface LocalStorageData {
+    readonly termsOfService: AcceptedTosData | null
+  }
+}
+
+LocalStorage.registerKey('termsOfService', {
+  tryParse: value => {
+    if (
+      typeof value === 'object' &&
+      value != null &&
+      'versionHash' in value &&
+      typeof value.versionHash === 'string'
+    ) {
+      // eslint-disable-next-line no-restricted-syntax
+      return value as localStorage.LocalStorageData['termsOfService']
+    } else {
+      return null
+    }
+  },
+})
+
+export const latestTermsOfService = reactQuery.queryOptions({
+  queryKey: ['termsOfService', 'currentVersion'],
+  queryFn: () =>
+    fetch(new URL('/eula.json', process.env.ENSO_CLOUD_ENSO_HOST))
+      .then(response => {
+        if (!response.ok) {
+          throw new Error('Failed to fetch terms of service')
+        } else {
+          return response.json()
+        }
+      })
+      .then(data => {
+        invariant(data != null, 'Invalid terms of service response')
+        invariant(
+          // This is safe because we've checked that `data` is not null and that it has a `hash` property
+          // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
+          'hash' in data && typeof data.hash === 'string',
+          'Invalid terms of service response'
+        )
+
+        // This is safe because we've checked the data above
+        // eslint-disable-next-line no-restricted-syntax
+        return data as { hash: string }
```

Comment:
unnecessary type assertion. return `{ hash }` instead

Reply:
Unfortunately, ts complains that `hash` has `any` type. Maybe I'm doing something wrong?

Reply:
sorry my bad, `{ hash: data.hash }`

Reply:
![CleanShot 2024-05-17 at 13 21 53@2x](https://github.com/enso-org/enso/assets/61194245/30e302d1-2819-438f-a753-b0d85aada75c)

Here's the what I got:

Reply:
Oh, awesome. I manually casted data to unknown